### ISSUE TITLE
Linker cannot find ghoul's zip library during make

### ISSUE THREAD

**Issue by JonisK** [OpenSpace/OpenSpace#1777]
I followed the [Ubuntu compile guide](http://wiki.openspaceproject.com/docs/developers/compiling/ubuntu) and got an error while executing the `make` command.

``` bash
[ 93%] Linking CXX executable ../../../../../bin/codegentest
/usr/bin/ld: ../../../../ext/spice/libspice.a(open.c.o): in function `f_open':
open.c:(.text+0x894): warning: the use of `mktemp' is dangerous, better use `mkstemp' or `mkdtemp'
/usr/bin/ld: ../../../../src/libopenspace-core.a(scriptengine.cpp.o): in function `openspace::luascriptfunctions::unzipFile(lua_State*)':
/home/jonis/git/OpenSpace/src/scripting/scriptengine_lua.inl:304: undefined reference to `zip_extract'
collect2: error: ld returned 1 exit status
make[2]: *** [support/coding/codegen/tests/CMakeFiles/codegentest.dir/build.make:943: ../bin/codegentest] Error 1
make[1]: *** [CMakeFiles/Makefile2:2836: support/coding/codegen/tests/CMakeFiles/codegentest.dir/all] Error 2
make: *** [Makefile:152: all] Error 2
```
The path that the script `scriptengine_lua.inl` references for the zip library looks good but then I am no expert at all in C. Copying ghoul's assimp-folder into the build directory did not help.

I am running Ubuntu 20.04 on an x86_64 machine.

Thanks in advance for pointing out what went wrong!

**Comment by GPayne** [comment]
@JonisK: thanks for sending your CMake output. The problem is that CMake is finding the libassimp installed in your OS and using that instead of using the provided code in ghoul ("_Using system assimp: libassimp.so.5_" in your output).

This happens because you have libassimp-dev installed in Ubuntu. (I have libassimp installed but not the -dev version). We will need to fix the CMake files to prevent it from using the system libassimp.

The temporary fix is to uninstall libassimp-dev from synaptic, unless you really need it.

**Comment by alexanderbock** [comment]
Another hotfix would be in here:
https://github.com/OpenSpace/Ghoul/blob/master/CMakeLists.txt#L170

Removing line 170 thorough 177 and 192  so that the `else` case is always executed.

@GPayne @micahnyc    Is there are reason why we'd want to ever use the built-in version on Linux or Mac?   On Windows, there isn't

**Comment by JonisK** [comment]
Hi @GPayne and @alexanderbock 

Thanks a lot for your suggestions! Indeed, I have libassimp-dev installed since it is a dependency for ROS noetic. 
I did not want to uninstall ROS on my machine so I went with alexander's suggestion to edit the CMakeLists.txt and now the project compiles successfully. Thanks again!

My takeaway from this issue is that ROS and OpenSpace are currently incompatible but if it would be possible to always disregard the preinstalled libassimp, like alexander suggests, then it should be no issue to have both installed at the same time.

Thanks again!
